modules/nextflow/src/main/groovy/nextflow/executor/TaskArrayExecutor.groovy

```diff
@@ -54,4 +57,31 @@ interface TaskArrayExecutor {
      */
     String getArrayTaskId(String jobId, int index)
 
+    default boolean isWorkDirDefaultFS() {
+        getWorkDir().fileSystem== FileSystems.default
+    }
+
+    /**
+     * Get a {@link TaskHandler} work directory for the task array resolution
+     *
+     * @param handler
+     * @return
+     */
+    default String getArrayWorkDir(TaskHandler handler) {
+        return isFusionEnabled()
+            ? FusionHelper.toContainerMount(handler.task.workDir).toString()
+            : handler.task.workDir.toUriString()
+    }
+
+    default String getArrayLaunchCommand(String taskDir) {
```

Comment:
While I understand the motivation to move this logic into the job array interface, I fear we will have to revert it once we implement task batching: https://github.com/nextflow-io/nextflow/pull/3909/files#diff-c317d610e9c87f9c425afaedfb79ed6ca92083197d0e0ac9be6b2299aab5860eR110-R123

I implemented `getWorkDir()` and `getLaunchCommand()` in the task handler specifically so that it can be used both for job arrays and task batching

Reply:
let's see this later